### ISSUE TITLE
Fail to retrieve / record the sleep hours after loading from saved sims

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#345]
**Describe the bug**
- Fail to record the sleep hours after loading from a saved sim

**Expected behavior**
- `CircadianClock` should record the sleep pattern

**Version**
 - Build 5705

**Additional context**

CircadianClock object was created in Person's initialize() as follows : 

```
	/**
	 * Initialize field data and class 
	 */
	public void initialize() {
		// Create ciracdian clock
		circadian = new CircadianClock(this);
        }
```

In CircadianClock, the map sleepCycleMap record the recent sleep pattern and the map and sleepTime stores the history of sleep for each person.

```
	/** Sleep habit map keeps track of the sleep cycle */
	private Map<Integer, Integer> sleepCycleMap;

	/** The amount of Sleep [millisols] a person on each mission sol */
	private Map<Integer, Double> sleepTime;
```

**Comment by mokun** [comment]
@bevans2000 , 

Do you think sometimes must have changed how the the two sleep related maps are serialized and thus are no longer available after loading from a saved sim ?

**Comment by bevans2000** [comment]
I could not find anything wrong with the sleep recording. However I have cleaned-up the code and removed the need to hold a back pointer to the parent Person object.. This may cause a serialization issue. I have moved the recording of the sleep time over to the DataLogger pattern since this simplifies the recording and daily roll over.

**Comment by bevans2000** [comment]
7f62e4fae2be07d9463aaf77594f048cef5638f8

**Comment by mokun** [comment]
@bevans2000 ,

Somehow, the sleep statistics is still NOT being recorded or loaded properly into the Health Tab in Person Window.

![image](https://user-images.githubusercontent.com/1168584/120937757-17cafa00-c6c4-11eb-9cfd-669e0e796a9a.png)

It shows that between sol 1 to sol 5, there are zero sleep hours. Nothing is recorded after sol 5.

**Comment by mokun** [comment]
And there is a separate issue with the time statistics. 

It shows wrong values in the Time Tool when the sim is supposed to be on pause. See below.

![image](https://user-images.githubusercontent.com/1168584/120938325-f6b7d880-c6c6-11eb-9853-21ee60f37bbc.png)

**Comment by bevans2000** [comment]
> @bevans2000 ,
> 
> In this sim session, I'm going all the way straight onto sol 29 and have NEVER reloaded it from a saved sim at all.
> 

Ok, so you let the initial sim run for over 5 sols ? I can have a look.

**Comment by mokun** [comment]
> Ok, so you let the initial sim run for over 5 sols ?

Yes the simulation was just over one month.

I ran into issues with the statistics on the Time Tool.

Guess it has to do with the pause state and I've made an attempt to fix it in that [commit](https://github.com/mars-sim/mars-sim/commit/72579b69ed93547b5788ec5a7c66ef06fa19db11).

Since then, whenever there is an exception on something, it would blame it on the clock listener because of the change.
